core/src/software/aws/toolkits/core/credentials/ToolkitCredentialsProvider.kt

```diff
@@ -38,13 +38,12 @@ abstract class ToolkitCredentialsProvider : AwsCredentialsProvider {
      * Returns true or throws an Exception
      */
     @Throws(Exception::class)
-    open fun isValidOrThrow(stsClient: StsClient) {
+    open fun getAwsAccount(stsClient: StsClient): String =
```

Comment:
this doesn't feel like we have the correct dependency model here - you shouldn't have to pass an instance of an sts client in in order to be able to satisfy the call to this method.

Reply:
We shouldn't, but the trick here is that this class `ToolkitCredentialsProvider` is in `core` module, while the toolkit user agent is in `jetbrains-core` module. We want to use the same toolkit user agent everywhere.